packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -17,30 +17,29 @@ export const preloadMethod = `__vitePreload`
 export const preloadMarker = `__VITE_PRELOAD__`
 
 const preloadHelperId = 'vite/preload-helper'
-const preloadCode = `let scriptRel;const seen = {};export const ${preloadMethod} = ${preload.toString()}`
 const preloadMarkerRE = new RegExp(`"${preloadMarker}"`, 'g')
 
 /**
  * Helper for preloading CSS and direct imports of async chunks in parallel to
  * the async chunk itself.
  */
+
+function detectScriptRel() {
+  // @ts-ignore
+  const relList = document.createElement('link').relList
+  // @ts-ignore
+  return relList && relList.supports && relList.supports('modulepreload')
+    ? 'modulepreload'
+    : 'preload'
+}
+
+declare const scriptRel: string
```

Comment:
Maybe we can add

```ts
declare const seen: Record<string, boolean>
declare const __VITE_IS_MODERN__: boolean
```

Reply:
All good ideas, I think you can create a PR with all these three after we merge this one 👍🏼